File: AccountErp.Web/ClientApp/src/models/invoice/invoice.edit.model.ts
import { AttachmentEditModel } from '../';

export class InvoiceEditModel {
    public id: number;
    public customerId: string;
    public phone: string;
    public email: string;
    public invoiceNumber: string;
    public tax: number;
    public discount: number;
    public totalAmount: number;
    public remark: string;
    public poso: number;
    public memo: string;
    public invoiceDate: string;
    public dueDate: string;

    public items: any;
    public attachments: Array<AttachmentEditModel>;

    constructor() {
        this.customerId = '';
        this.items = new Array<number>();
        this.attachments = new Array<AttachmentEditModel>();
    }
}

File: AccountErp.Models/Invoice/InvoiceEditModel.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace AccountErp.Models.Invoice
{
    public class InvoiceEditModel
    {
        public int Id { get; set; }

        [Required]
        public int CustomerId { get; set; }
        
        public decimal? Tax { get; set; }

        public decimal? Discount { get; set; }

        [Required]
        public decimal TotalAmount { get; set; }
        public string Remark { get; set; }
        public DateTime InvoiceDate { get; set; }
        public DateTime DueDate { get; set; }
        public decimal? PoSoNumber { get; set; }
        public decimal? SubTotal { get; set; }

        [Required]
        public List<InvoiceServiceModel> Items { get; set; }
        //public List<int> Items { get; set; }

        public List<InvoiceAttachmentAddModel> Attachments { get; set; }
    }
}
